client/state/data-layer/README.md

```diff
@@ -56,15 +56,14 @@ Instead the components should be able to request that they require such data and
 
 ## Implementation
 
-The data layer _intercepts_ Redux actions.
-Once it has performed its behavior, this action will be dropped entirely from the dispatch path unless explicitly forwarded to the next middleware or dispatched again.
-This is to prevent two middlewares which can both supply a given request from fighting with each other or duplicating the request; consequently it allows for prioritization of data-fetching needs by means of ordering how the middleware are arranged in the chain.
-For example, we could start polling from the WP-API for posts but leave all other requests up to the WordPress.com API.
-Note that we can still allow multiple functions within the same middleware to handle the same action.
-The WordPress.com API middleware demonstrates this in building a tree of handlers which can all respond to given action types.
+The data layer _responds to_ Redux actions.
+Once it has performed its behavior, the original action will continue along the dispatch path to the next middleware in the chain.
+We can create multiple "handler" functions for the same action type and they will be triggered in the order they are registered when that action type dispatches.
+The WordPress.com API middleware demonstrates this in building a tree of handlers which can all respond to given data requests.
 
-Each middleware intercepts given Redux actions and will correspondingly dispatch new follow-up actions to actually handle their requests.
-The functions that compose to form the middleware _can_ and in most cases _will_ closely resemble what was previously written in `redux-thunk` actions.
+Each middleware handler can dispatch new follow-up actions to respond in some way to the original action.
+For example, they will often dispatch a new HTTP request.
+The functions that compose to form the middleware _can_ and in most cases _will_ closely resemble what was previously written in `redux-thunk` actions but in this system they will be controllable.
```

Comment:
updated! I didn't want to go too deep here but I did add some links on the subject matter in case anyone has the inclination…